templates/CRM/common/TrackingFields.tpl

```diff
@@ -7,7 +7,7 @@
  | and copyright information, see https://civicrm.org/licensing       |
  +--------------------------------------------------------------------+
 *}
-{if $trackingFields and ! empty($trackingFields)}
+{if array_key_exists('trackingFields', $form) and !empty($trackingFields)}
```

Comment:
Does this make sense?
Also I can't find where trackingFields is ever defined?

Reply:
It was an old 'good idea' by Lobo - I was tempted to just pull them out & see if anyone noticed ... 

https://civicrm.org/blog/lobo/tracking-contributions-and-event-registrations-for-campaigns-and-appeals

Reply:
I didn't find any universe refs & apparently Wikimedia was the original 'client'

Reply:
The case for removing them from message templates seems very strong....

Reply:
I had to implement this, this week. I didn't know about `$trackingFields`, but I did pass values via the URL.

`$trackingFields` hides the fields from the form and receipt. My client was OK with the fields being visible and they use custom receipts. Otherwise, if I knew about this feature, I'd have used it.

I once proposed custom fields of type "hidden", which would solve this and more.

Thinking ahead - FormBuilder can populate fields from the URL without displaying them. Though I doubt anyone's figured out receipts when FB can process payments...

Reply:
@MegaphoneJon I guess that suggests that maybe someone else might use them if they are useful.

Reply:
Also - wouldn't it be a case of adding `is_hidden` to the field - which then starts to ask if we should use `visibility` better ....